src/mmw/apps/modeling/geoprocessing.py

```diff
@@ -71,6 +72,21 @@ def sjs_submit(host, port, args, data, retry=None):
             except MaxRetriesExceededError:
                 raise Exception('No slots available in Spark JobServer.\n'
                                 'Details = {}'.format(response.text))
+        elif error['result'] == 'context geoprocessing not found':
+            geop_context_url = '{}/contexts?reset=reboot'.format(base_url)
+            context_response = requests.put(geop_context_url)
+
+            if context_response.ok:
+                if retry:
+                    retry()
+                else:
+                    raise Exception('Geoprocessing context missing in Spark'
+                                    'JobServer, but no retry was set.\n'
+                                    'Details = {}'.format())
+            else:
+                raise Exception('Unable to create missing geoprocessing'
```

Comment:
You may need a space at the end of `geoprocessing` else it will be joined with `context` on the next line:

```python
In [1]: print('Unabe to create missing geoprocessing'
   ...: 'context in Spark JobServer.\n')
Unabe to create missing geoprocessingcontext in Spark JobServer.

```